### PR TITLE
Update to new YieldMath

### PR DESCRIPTION
**Pull request by Sabnock01**
#77 should probably be merged first. That version of the npm package reflects the updates to the `PoolOracle`.

### CODE REVIEW

File: src/YieldMath.sol [hunk 1]
```diff
@@ -447,7 +447,9 @@ library YieldMath {
                 (uint128(zaYaYxa.divu(uint128(c.div(mu)))).pow(uint128(ONE), uint128(a))).i128()
             );
 
-            return uint128(subtotal) - sharesReserves;
+            uint128 sharesOut = uint128(subtotal) - sharesReserves;
+            require(sharesOut < uint128(subtotal), "YieldMath: Underflow error");
```
Comment (alcueca) [inline src/YieldMath.sol:451]: What if `sharesReserves` is zero?

Comment (alcueca) [reply src/YieldMath.sol:451]: Probably something else would revert, but still I think that's a `<=`

File: src/YieldMath.sol [hunk 2]
```diff
@@ -573,6 +575,7 @@ library YieldMath {
 
             // maxFYTokenOut_ = fyTokenReserves - (rightTerm * 1e18)
             require((fyTokenOut = fyTokenReserves - uint128(rightTerm.mulu(WAD))) <= MAX, "YieldMath: Underflow error");
+            require(fyTokenOut < fyTokenReserves, "YieldMath: Underflow error");
```
Comment (alcueca) [inline src/YieldMath.sol:578]: Likewise

File: src/YieldMath.sol [hunk 3]
```diff
@@ -627,6 +630,7 @@ library YieldMath {
 
             // maxSharesIn_ = (leftTerm * 1e18) - sharesReserves
             require((sharesIn = uint128(leftTerm.mulu(WAD)) - sharesReserves) <= MAX, "YieldMath: Underflow error");
+            require(sharesIn < uint128(leftTerm.mulu(WAD)), "YieldMath: Underflow error");
```
Comment (alcueca) [inline src/YieldMath.sol:633]: Likewise

**Review by alcueca — CHANGES_REQUESTED** [review]
I always have something to say :D

File: src/YieldMath.sol (hunk 1 above)
Comment (Sabnock01) [reply src/YieldMath.sol:451]: You're right.

**Review by alcueca — APPROVED** [review]
LGTM

File: src/YieldMath.sol [hunk 4]
```diff
@@ -447,7 +447,9 @@ library YieldMath {
                 (uint128(zaYaYxa.divu(uint128(c.div(mu)))).pow(uint128(ONE), uint128(a))).i128()
             );
 
-            return uint128(subtotal) - sharesReserves;
+            uint128 sharesOut = uint128(subtotal) - sharesReserves;
```
Comment (devtooligan) [inline src/YieldMath.sol:450]: Please add this comment:
```suggestion
            // subtotal is calculated as a positive fraction multiplied by a uint so it cannot underflow when casting to uint and its ok to use a raw casting
            uint128 sharesOut = uint128(subtotal) - sharesReserves;
```

**Pull request merged**